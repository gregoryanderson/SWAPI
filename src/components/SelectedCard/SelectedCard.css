.selected__div {
    align-items: center;
    width: 30%;
    height: 300px;
    margin: 0px 5px;
    border-radius: 5px;
    background-image: url('https://png.pngtree.com/thumb_back/fh260/background/20190510/pngtree-starry-sky-white-blue-stars-image_111634.jpg');
    background-size: cover;
    border: 3px solid black;
  }
  
