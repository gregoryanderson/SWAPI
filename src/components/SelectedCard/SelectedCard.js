import React from 'react';
import { Link } from 'react-router-dom';
import './SelectedCard.css';
import PropTypes from 'prop-types'

const SelectedCard = ({ data }) => {
  const {name, type} = data
    return (
      <div className="selected__div">
        <Link to={`/${type}`} className='back-btn'>â—€ back</Link>
        <header className="selected__header">
          <h3 className="selected__h3">{name}</h3>
          <img src='http://images2.wikia.nocookie.net/__cb20080228205028/starwars/images/thumb/7/71/Redstarbird.svg/1600px-Redstarbird.svg.png' height='40px' width='40px' alt='Star Wars Favorite Button'/>
        </header>
      </div>
    )
  }

export default SelectedCard;

SelectedCard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string, 
  type: PropTypes.string
}