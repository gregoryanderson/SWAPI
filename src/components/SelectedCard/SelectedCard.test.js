import React from "react";
import { shallow } from "enzyme";
import SelectedCard from './SelectedCard';
import { exportAllDeclaration } from "@babel/types";

describe("SelectedCard", () => {
    it("should match the snapshot with all the data passed through", () => {
        const wrapper= shallow(<SelectedCard 
            id= {4}
            name={'Darth Vader'}
            type = {'people'}
            />)
        expect(wrapper).toMatchSnapshot();
    })
})