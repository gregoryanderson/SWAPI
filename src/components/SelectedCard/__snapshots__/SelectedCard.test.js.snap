// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedCard should match the snapshot with all the data passed through 1`] = `
<div
  className="selected__div"
>
  <Link
    className="back-btn"
    to="/people"
  >
    â—€ back
  </Link>
  <header
    className="selected__header"
  >
    <h3
      className="selected__h3"
    >
      Darth Vader
    </h3>
    <img
      alt="Star Wars Favorite Button"
      height="40px"
      src="http://images2.wikia.nocookie.net/__cb20080228205028/starwars/images/thumb/7/71/Redstarbird.svg/1600px-Redstarbird.svg.png"
      width="40px"
    />
  </header>
</div>
`;
